addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -6598,15 +6598,17 @@ registry.ImageTools = ImageHandlerOption.extend({
             mimetype: this._getImageMimetype(img),
         });
 
-        await new Promise(resolve => {
-            this.$target.one('image_cropper_destroyed', async () => {
-                if (isGif(this._getImageMimetype(img))) {
-                    img.dataset[img.dataset.shape ? 'originalMimetype' : 'mimetype'] = 'image/png';
-                }
-                await this._reapplyCurrentShape();
-                resolve();
+        if (!imageCropWrapper.component._cropperClosed) {
+            await new Promise(resolve => {
+                this.$target.one("image_cropper_destroyed", async () => {
+                    if (isGif(this._getImageMimetype(img))) {
+                        img.dataset[img.dataset.shape ? "originalMimetype" : "mimetype"] = "image/png";
+                    }
+                    await this._reapplyCurrentShape();
+                    resolve();
+                });
             });
-        });
+        }
```

Comment:
It's not nice to have to read internal values of the component (`.component._cropperClosed`). If I understand this well, the cropper is immediately destroyed, hence the event listener was not attached yet? Should not the fix be about just moving the the listener addition before attaching the component?